core/capability-negotiation.md

```diff
@@ -0,0 +1,437 @@
+    Server: CAP * LS :draft/example-1 draft/example-2
+    Client: CAP END
+    Server: 001 dan :Welcome to the Internet Relay Network dan
+    ...
+
+Example without capability negotiation:
+
+    Client: CAP LS 302
+    Client: NICK dan
+    Client: USER d * 0 :This is a really good name
+    Server: 001 dan :Welcome to the Internet Relay Network dan
+    ...
+
+## The CAP command
+
+The client capability negotiation extension is implemented by the addition of one command
+with several subcommands.  The command added is named `CAP`.  `CAP` takes a single
+required subcommand, optionally followed by a single parameter of space-separated capability
+identifiers.  Each capability in the list may be preceded by a capability modifier.
+
+The subcommands for `CAP` are: `LS`, `LIST`, `REQ`, `ACK`, `NAK`, and `END`.
+
+The `LS`, `LIST`, `REQ`, `ACK` and `NAK` subcommands may be followed by a single parameter
+containing a space-separated list of capabilities.  If more than one capability is named,
+the RFC1459 designated sentinel (`:`) for a multi-parameter argument must be present. The
+list of capabilities MUST be parsed from left to right and capabilities SHOULD only be sent
+once per command. If a capability is sent multiple times, the last one received takes priority.
+
+If a client sends a subcommand which is not in the list above or otherwise issues an
+invalid command, then numeric 410 (ERR_INVALIDCAPCMD) should be sent.  The first parameter
+after the client identifier (usually nickname) should be the commandname; the second parameter
+should be a human-readable description of the error.
+
+Replies from the server must contain the client identifier name or asterisk if one is not yet
+available.
+
+Example:
+
+    Client: CAP FOO
+    Server: :example.org 410 * FOO :Invalid CAP command
+
+The client MUST be able to use the `CAP` command anytime, even after registration.
+
+### The CAP LS subcommand
+
+The LS subcommand is used to list the capabilities supported by the server.  The client
+should send an LS subcommand with no other arguments to solicit a list of all capabilities.
+
+If a client issues an LS subcommand, registration must be suspended until an END subcommand
+is received.  If no capabilities are available, an empty parameter must be sent.
+
+Example:
+
+    Client: CAP LS
+    Server: CAP * LS :multi-prefix sasl
+
+#### CAP LS Version
+
+The LS subcommand has an additional argument which is the version number of the latest
+capability negotiation protocol supported by the client.
+
+Clients that send `302` as the CAP LS version are presumed to support `CAP LS 302` features
+for the future life of the connection.
+
+Clients that do not send any version number with CAP LS are presumed to not support these
+extra features.
+
+If a client has not indicated support for `CAP LS 302` features, the server MUST NOT send
+these new features to the client such as cap values and multiline replies.
```

Comment:
I think a complete summary of 302 features might make sense here. As written you need to skim the rest of the document for references to 302.

Reply:
I've added a table listing the features just below. Once it gets in I'll see how the names get added on the site and link to them from this table.